editor/sidebar/featured-image/index.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { __ } from 'i18n';
+import Button from 'components/button';
+import PanelBody from 'components/panel/body';
+import MediaUploadButton from 'blocks/media-upload-button';
+import Spinner from 'components/spinner';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class FeaturedImage extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			media: null,
+			loading: false,
+		};
+
+		// This boolean should be removed when we provide QueryComponents
+		this.mounted = true;
+	}
+
+	componentDidMount() {
+		this.fetchMedia();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.featuredImageId !== this.props.featuredImageId ) {
+			this.fetchMedia();
+		}
+	}
+
+	componentWillUnmount() {
+		this.mounted = false;
+	}
+
+	fetchMedia() {
+		if ( ! this.props.featuredImageId ) {
+			this.setState( {
+				media: null,
```

Comment:
Maybe moving up the common `media: null` would make it more reasonable to turn setters into one-liners to trim a few lines and clarify the common behavior.

```js
this.setState( { media: null } );
if ( ! this.props.featuredImageId ) {
	this.setState( { loading: false } );
	return;
}
this.setState( { loading: true } );
```

It's unclear from [the documentation](https://facebook.github.io/react/docs/react-component.html#setstate) whether the current implementation would cause the render to occur twice by separate calls to `setState`, so maybe worth avoiding if it does.